File: app/src/main/java/com/wengtao/my_utils/Command.java
package com.wengtao.my_utils;

import java.io.ByteArrayOutputStream;
import java.io.InputStream;

/**
 * Created by wt on 2017-1-6.
 */

public class Command {
    public static String execCmd(String... command) {
        Process process;
        InputStream errIs, inIs;
        String result;
        try {
            process = new ProcessBuilder().command(command).start();
            ByteArrayOutputStream baos = new ByteArrayOutputStream();
            int read;
            errIs = process.getErrorStream();
            while ((read = errIs.read()) != -1) {
                baos.write(read);
            }

            inIs = process.getInputStream();
            while ((read = inIs.read()) != -1) {
                baos.write(read);
            }

            result = new String(baos.toByteArray());

            errIs.close();
            inIs.close();
        } catch (Exception e) {
            result = e.toString();
        }
        return result;
    }
}

File: app/src/main/java/com/wengtao/my_utils/Command.kt
package com.wengtao.my_utils

import java.io.ByteArrayOutputStream
import java.io.InputStream

public class Command {
    val TAG = "Command"


    companion object {
        fun execCmd(vararg commands: String): String? {
            val process: Process
            val errIs: InputStream
            val inIs: InputStream
            var result: String? = null
            if (commands.isNotEmpty()) {
                try {
                    process = ProcessBuilder().command(*commands).start()
                    val byteArrayOutputStream = ByteArrayOutputStream()
                    var read = 0
                    errIs = process.errorStream
                    do {
                        read = errIs.read()
                        byteArrayOutputStream.write(read)
                    } while (read != -1)

                    inIs = process.inputStream

                    do {
                        read = inIs.read()
                        byteArrayOutputStream.write(read)
                    } while (read != -1)

                    result = String(byteArrayOutputStream.toByteArray())
                    errIs.close()
                    inIs.close()
                } catch (e: Exception) {
                    e.printStackTrace()
                }
            }
            return result
        }
    }


}